app/code/Magento/MediaGallery/Plugin/Wysiwyg/Images/Storage.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\MediaGallery\Plugin\Wysiwyg\Images;
+
+use Magento\MediaGalleryApi\Model\Asset\Command\GetByPathInterface;
+use Magento\MediaGalleryApi\Model\Asset\Command\DeleteByPathInterface;
+use Magento\Cms\Model\Wysiwyg\Images\Storage as StorageSubject;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Exception\ValidatorException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Ensures that metadata is removed from the database when a file is deleted and it is an image
+ */
+class Storage
+{
+    /**
+     * @var GetByPathInterface
+     */
+    private $getMediaAssetByPath;
+
+    /**
+     * @var DeleteByPathInterface
+     */
+    private $deleteMediaAssetByPath;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Storage constructor.
+     *
+     * @param GetByPathInterface $getMediaAssetByPath
+     * @param DeleteByPathInterface $deleteMediaAssetByPath
+     * @param Filesystem $filesystem
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        GetByPathInterface $getMediaAssetByPath,
+        DeleteByPathInterface $deleteMediaAssetByPath,
+        Filesystem $filesystem,
+        LoggerInterface $logger
+    ) {
+        $this->getMediaAssetByPath = $getMediaAssetByPath;
+        $this->deleteMediaAssetByPath = $deleteMediaAssetByPath;
+        $this->filesystem = $filesystem;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Delete media data after the image delete action from Wysiwyg
+     *
+     * @param StorageSubject $subject
+     * @param StorageSubject $result
+     * @param string $target
+     *
+     * @return StorageSubject
+     * @throws ValidatorException
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterDeleteFile(StorageSubject $subject, StorageSubject $result, $target): StorageSubject
+    {
+        if (!is_string($target)) {
+            return $result;
+        }
+
+        $relativePath = $this->filesystem->getDirectoryRead(DirectoryList::MEDIA)->getRelativePath($target);
+        if (!$relativePath) {
+            return $result;
+        }
+
+        try {
+            $this->deleteMediaAssetByPath->execute($relativePath);
+        } catch (\Exception $exception) {
+            $message = __('An error occurred during media asset delete at wysiwyg: %1', $exception->getMessage());
+            $this->logger->critical($message->render());
```

Comment:
What the reason to write a localized message to the log instead of original exception?

Reply:
Hi @joni-jones
The __() decided to be used here to avoid direct usage of the sprintf. The text before error message really helps during the issue identification, debug and especially useful when you use it as a key for log filtering. If I am not mistaken, the error message will be in a default locale at any case for plugins (will re-check it).

Reply:
For issue identification, an exception provides a stack trace that contains the chain of code execution. In the current solution, you will see the error message which happens in wysiwyg but without the whole picture.

Reply:
@joni-jones thank you for your answer. I made some tests: added an undefined variable at delete command and run the plugin. Got the next results. Please, check them when you will have a time:

1. This is for the current implementation:
`[2019-11-06 12:32:59] main.CRITICAL: An error occurred during media asset delete at WYSIWYG: Notice: Undefined variable: a in /Volumes/Projects/apache/m23/ext/adobe-stock-integration/AdobeMediaGallery/Plugin/Wysiwyg/Images/Storage.php on line 86 [] []`

2. This is for `$this->logger->critical($exception->getMessage());`
`[2019-11-06 12:36:25] main.CRITICAL: Could not delete media asset with path wysiwyg/-.jpeg: Notice: Undefined variable: a in /Volumes/Projects/apache/m23/ext/adobe-stock-integration/AdobeMediaGallery/Model/Asset/Command/DeleteByPath.php on line 64 [] []`

3. This is for `$this->logger->critical($exception);`
`[2019-11-06 12:57:54] main.CRITICAL: Could not delete media asset with path wysiwyg/cold-sheep.jpeg: Notice: Undefined variable: a in /Volumes/Projects/apache/m23/ext/adobe-stock-integration/AdobeMediaGallery/Model/Asset/Command/DeleteByPath.php on line 64 [] []`

It looks like in all variants we have desired result: an error description. The trace usually locates in the report log. The exception has it in the message which we write to log.  Or this test is invalid?

Reply:
@coderimus Considering that there can be a different implementation of the logger, an exception can be logged in different ways, but I think it's better to pass to the logger as much information as we can. Fixed.

Reply:
@coderimus your test is good enough, results are mostly the same because of Magento implementation `\Magento\Framework\Logger\Monolog::addRecord` for Monolog. Currently, you will see the only trace line where actually the exception is caught. But Monolog it's just an implementation used Magento by default and there can be any logging mechanism compatible with PSR Logger, as @sivaschenko mentioned, it's better to not limit logging possibilities and that was my main point.

Reply:
@joni-jones @sivaschenko thank you for explanation. Based on it I will re-check Adobe Stock Integration modules too and fix logging where possible.